astropy/table/operations.py

```diff
@@ -129,6 +130,86 @@ def join(left, right, keys=None, join_type='inner',
     return out
 
 
+def setdiff(table1, table2, keys=None):
+    """
+    Take a set difference of table rows.
+
+    The row set difference will contain all rows in table1 that are not
+    present in table2.
+
+    Parameters
+    ----------
+    table1 : Table object
+        table1 is on the left side of the set difference.
+    table2 : Table object
+        table2 is on the right side of the set difference.
+    keys : str or list of str
+        Name(s) of column(s) used to match rows of left and right tables.
+        Default is to use all columns which are common to both tables.
```

Comment:
This is different than what the code is doing, which is to use the columns of `table1`.  I actually think what the code is doing is better since basically `table1` is the "master" table for reference.